### ISSUE TITLE
TF::AzureAD::Application missing `single_page_application`

### ISSUE THREAD

**Issue by andreialecu** [iann0036/cfn-tf-custom-types#7]
Azure now has a special `single_page_application` type, which doesn't seem to be directly documented here:
https://github.com/iann0036/cfn-tf-custom-types/blob/docs/resources/azuread/TF-AzureAD-Application/docs/README.md

It's been added a while back:
https://github.com/hashicorp/terraform-provider-azuread/pull/474

I'm not sure what would need to be done here, I assume there's a script that could be used to re-generate everything?

**Comment by iann0036** [comment]
Yeah, it's in the README - per https://github.com/iann0036/cfn-tf-custom-types?tab=readme-ov-file#generation

**Comment by andreialecu** [comment]
Understood, do you plan on updating them in the main release as well? 

---

PS: While running `python3 generate.py azuread` I first got:

```
subprocess.CalledProcessError: Command '['cfn', 'init', '--type-name', 'TF::AzureAD::SynchronizationSecret', '--artifact-type', 'RESOURCE', 'python37', '--use-docker']' returned non-zero exit status 1
... and many more
```

I manually ran `cfn init` and saw 
```
➜ cfn init
ModuleNotFoundError: No module named 'pkg_resources'
```

For which I ran `pip install setuptools` and that fixed it.

Now getting:

```
subprocess.CalledProcessError: Command '['cfn', 'generate']' returned non-zero exit status 1.
... and many more
```

The error seems to be:

```
[2024-02-27T12:51:37Z] DEBUG    - Loading project file '/Users/andreialecu/Work/contrib/cfn-tf-custom-types/resources/azuread/TF-AzureAD-Application/.rpdk-config'
[2024-02-27T12:51:37Z] DEBUG    - Project file /Users/andreialecu/Work/contrib/cfn-tf-custom-types/resources/azuread/TF-AzureAD-Application/.rpdk-config not found. Have you run 'init' or in a wrong directory?
```

**Comment by andreialecu** [comment]
With the updates in #8 I was able to generate and submit the updated resource. However, it would be really great if you could update the main ones as well, because various cdk constructs, like: https://constructs.dev/packages/@cdk-cloudformation/tf-azuread-application/v/1.0.0-alpha.7?lang=typescript are being automatically generated from it.
